nixos/doc/manual/release-notes/rl-2009.xml

```diff
@@ -86,6 +86,53 @@
 }</programlisting>
     </para>
    </listitem>
+
+   <listitem>
+     <para>
+       The PHP package is now a much smaller package with defaults that are much closer to upstream
+       defaults for which extensions that should be enabled. Notable extensions that are disabled now
+       that used to be enabled contain among other things database related extensions.
+     </para>
+     <para>
+       Since this release we have an easy way to customize your PHP install to get
+       additional extensions enabled. See following snippet to install a PHP with <literal>gd</literal>,
+       <literal>pdo_mysql</literal> and <literal>pdo_pgsql</literal> added:
+
+       <programlisting>
+         environment.systemPackages = [
+           (pkgs.php.buildEnv { exts = pp: with pp.exts; [
+             gd
+             pdo_mysql
+             pdo_pgsql
+           ]; })
+         ];
+       </programlisting>
```

Comment:
Would it make sense to provide e.g. a `phpMinimal` or a `phpExtended` (with opinionated extensions) or even a `phpFull`?

Reply:
I would say that the one I provide in this PR is a good enough base, and if you want to build a minimal one you can do that. And a full one would be to add all the extensions to get a full one. And that shouldn't require any compiling so it's fast to build.

And I find the one I provide by default in this PR small enough so we don't need a minimal one in my opinion.

I was thinking about it before though, that's why the minimal flag is there at all :)

But the minimal one has some issues to build it completely minimal. That is if you have a minimal build without `--enable-mysqlnd` you're not able to build `phpPackages.exts.pdo_mysql` because the php used that provides `phpize` needs to be built with `--enable-mysqlnd` to have `mysqlnd.h` available.

This problem could be solved to use a less minimal php to build the extensions and then use a minimal as base for the package used. But then you would end up with both of those in the closure so I'd rather avoid that... :-)